gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionTransformer.scala

```diff
@@ -18,6 +18,15 @@ package org.apache.gluten.expression
 
 import org.apache.gluten.substrait.expression.ExpressionNode
 
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.types.DataType
+
 trait ExpressionTransformer {
```

Comment:
Does this one still should be used? Can we use `ExpressionTransformerWithOrigin` to replace `ExpressionTransformer` and rename `ExpressionTransformerWithOrigin` to `ExpressionTransformer`?

Reply:
Some expression transformer did not catch orginal expression. e.g., `BoundReferenceTransformer`, `AttributeReferenceTransformer`, etc. So I leave it in this pr.

Reply:
We can change it to catch original, do you have other example?
```
val b = bindReference.asInstanceOf[BoundReference]
          AttributeReferenceTransformer(
            a.name,
            b.ordinal,
            a.dataType,
            b.nullable,
            a.exprId,
            a.qualifier,
            a.metadata)

      case b: BoundReference =>
        BoundReferenceTransformer(b.ordinal, b.dataType, b.nullable)
```

Reply:
For the code clean, I prefer to hold origin for all the `ExpressionTransformer`, what do you think? @zhztheplayer